src/services/ledger/domain/entry-builder.ts

```diff
@@ -1,194 +1,195 @@
-    }) as EntryBuilderCredit<M, T>
+    }) as EntryBuilderCredit<M>
   }
 
-  const debitLnd = (amount: BtcPaymentAmount): EntryBuilderCreditWithBtcDebit<M> => {
-    entry.debit(lndLedgerAccountId, Number(amount.amount), {
-      ...metadata,
-      currency: amount.currency,
-    })
-    return EntryBuilderCreditWithBtcDebit({
-      entry,
-      metadata,
-      fee,
-      debitAmount: amount as BtcPaymentAmount,
-      staticAccountIds,
-    }) as EntryBuilderCreditWithBtcDebit<M>
+  const debitLnd = (): EntryBuilderCredit<M> => {
+    return debitAccount({ accountDescriptor: lndLedgerAccountDescriptor })
+  }
+
+  const debitColdStorage = (): EntryBuilderCredit<M> => {
+    return debitAccount({ accountDescriptor: coldStorageAccountDescriptor })
   }
 
   return {
     debitAccount,
     debitLnd,
+    debitColdStorage,
   }
 }
 
-type EntryBuilderCreditState<M extends MediciEntry, D extends WalletCurrency> = {
-  entry: M
-  metadata: TxMetadata
-  fee: BtcPaymentAmount
-  debitAmount: PaymentAmount<D>
-  staticAccountIds: {
-    dealerBtcAccountId: LedgerAccountId
-    dealerUsdAccountId: LedgerAccountId
-  }
-}
-
-const EntryBuilderCreditWithUsdDebit = <M extends MediciEntry>({
+const EntryBuilderCredit = <M extends MediciEntry>({
   entry,
   metadata,
-  debitAmount,
+  amountWithOutFee: { usdWithOutFee, btcWithOutFee },
+  amountWithFee: { usdWithFee, btcWithFee },
+  debitCurrency,
   staticAccountIds,
-  fee,
-}: EntryBuilderCreditState<M, "USD">): EntryBuilderCreditWithUsdDebit<M> => {
-  const creditLnd = (btcCreditAmount: BtcPaymentAmount) => {
-    withdrawUsdFromDealer({
-      entry,
-      metadata,
-      staticAccountIds,
-      btcAmount: btcCreditAmount,
-      usdAmount: debitAmount,
-    })
-    const creditAmount = calc.sub(btcCreditAmount, fee)
-    entry.credit(lndLedgerAccountId, Number(creditAmount.amount), {
-      ...metadata,
-      currency: btcCreditAmount.currency,
-    })
-    return entry
-  }
-  const creditAccount = ({
-    accountId,
-    amount,
-  }: {
-    accountId: LedgerAccountId
-    amount?: BtcPaymentAmount
-  }) => {
-    if (amount) {
-      withdrawUsdFromDealer({
+}: EntryBuilderCreditState<M>): EntryBuilderCredit<M> => {
+  const creditLnd = () => creditAccount(lndLedgerAccountDescriptor)
+  const creditColdStorage = () => creditAccount(coldStorageAccountDescriptor)
+
+  const creditAccount = <T extends WalletCurrency>(
+    accountDescriptor: LedgerAccountDescriptor<T>,
+  ) => {
+    if (
+      debitCurrency === WalletCurrency.Usd &&
+      accountDescriptor.currency === WalletCurrency.Usd &&
+      usdWithFee.amount != usdWithOutFee.amount
+    ) {
+      const dealerData = {
         entry,
         metadata,
         staticAccountIds,
-        btcAmount: amount,
-        usdAmount: debitAmount,
-      })
+        btcAmount: calc.sub(btcWithFee, btcWithOutFee),
+        usdAmount: calc.sub(usdWithFee, usdWithOutFee),
+      }
+      withdrawUsdFromDealer(dealerData)
```

Comment:
this is a bad practice (it is really difficult to follow what is the purpose of the method) or we need to change the method name or we need to update entry in this line.

Reply:
@bodymindarts This is Juan's only hold up with the PR. I believe the concern is that when we are using a separate method (now named ```AddUsdToBtcConversionToEntry```)  to make changes to the entry, we are obscuring that the entry is being changed. @dolcalmi by returning the entry from ```AddUsdToBtcConversionToEntry``` and changing the name, does this address your problem?

Reply:
I think I understand @dolcalmi comment and think that the renaming is the easiest way to address it. I wouldn't use a capitalised name however (as this typically indicates a kind of constructor). `addUsdToBtcConvertionSoEntry` seems good however.